app/models/concerns/obstructed.rb

```diff
@@ -0,0 +1,70 @@
+# In queen.rb, rook.rb, bishop.rb, pawn.rb, I added "include Obstructed"

```

Comment:
Haha alright I was really trying to make this work, but it is pretty unconventional so I suppose it would be easier method in the Piece model.